### DISCUSSION TITLE
Some thoughts for enhancements.

### DISCUSSION THREAD

**Discussion by Wings64** [nguyenquyhy/Flight-Tracker-StreamDeck#231] [Ideas]
Hi,

I've almost finished finalising an FS2020  Caravan set for the XL Stream Deck.

Would it be possible to modify the generic gauge to allow custom colour bands as with the custom gauge? I'd like to show speed bands on a  'circular' analogue airspeed indicator. It which works just fine without, but would benefit from that final polish if possible to indicate stall, flap, and maximum speeds.

Also, is there a special way to change the background colour on the various buttons? If I do that in the usual way, I lose the readout of value. Reason is I'm trying to have the same background colour for members of a group to help separate the different functional groups.

The only way I have found to show the state of the parking brake is to use a generic toggle button to show BRAKE_PARKING_POSITION  as 0 or 1. Custom images for the states don't show, but remove the numbers. Is there a better way?

Thanks as always. Great plugin.

David J

**Comment by rmroc451** [comment]
You can embed backgrounds on all buttons, scroll to the bottom of the properties and you can set them.

Putting it on the streamdeck level overwrites what we write to the button :)

I've wanted to get to the custom colors for the generic circular style gauge, just haven't had the time :)

On the parking break stuff, try adding ==1 in the feedback so it evaluates to a boolean.

**Comment by Wings64** [comment]
Thanks for the speedy reply.

 

I appreciate, as I said, that you can change the background of buttons. The problem is that if you do that you no longer see the numeric readout that can be shown by populating the ‘Display Value’ field.

 

David J

**Reply by rmroc451** [reply]
Yeah you can! It sounds like you are applying the background to the button via the built in streamdeck option on the left hand side.

If you use the options/settings section on our buttons instead (all the way at the bottom) it will apply the image behind all the text that we spit out!

Can you check your implementation and let me know?

We had someone else have this same issue on our MSFS forum post this last week too :)

**Comment by Wings64** [comment]
I’m obviously missing something.

 

The Generic Toggle Buttons only offer on/off image fields, which I haven’t managed to get working for parking brake as I described.

 

The Generic Gauge doesn’t offer an image option.

 

Where should I be looking?

 

David J

**Reply by rmroc451** [reply]
Oh, you were talking the generic gauge background...  we write that yellow arc as a background, so you cant replace your own currently.

I mean, maybe we can open that up to let folks override that and copy the image that comes with the plugin (potentially).

I'll whip up an implementation of the parking brake and attach it here, when I get back to my PC today :)

**Comment by Wings64** [comment]
A decent parking brake button would be very helpful. Thanks !

 

Custom bgnd option for the generic button would be very helpful both to help group gauges and to allow banded segments. Adding a 288 square template would save automating like the custom straight gauge.

 

Thanks again.

**Reply by rmroc451** [reply]
[Default Profile.zip](https://github.com/nguyenquyhy/Flight-Tracker-StreamDeck/files/6255224/Default.Profile.zip)
Here is my test profile for parking brake. Give the settings on the buttons a look.

They all have the same feedback/toggle info, but I embedded all my different background types for them :)

**Comment by Wings64** [comment]
Excellent. Thank you very much. They all work just fine and you have also shown me how to use a 0/1 parameter as true/false.

 

At the risk of pushing my luck, could you also please explain how the Unit box should be populated and to what effect. I find the documentation rather silent in that area.

 

David

**Reply by rmroc451** [reply]
You can specify a unit that simconnect knows about and it will try to use that to spit the value out converted to that.

Like indicated_altitude, you could use meters.  Left_fuel_quantity, you could use litres. So on and so on.

I am again away from my pc, but a cursory google search for simconnect units yielded this:

https://www.prepar3d.com/SDKv4/sdk/references/variables/units_of_measurement.html

**Comment by Wings64** [comment]
Ryan,

 

Please forgive the further intrusion.

 

I am trying to get a real-world torque readout for the Caravan in FS2020.

 

The only torque parameter in the lists I have is TURB_ENG_MAX_TORQUE_PERCENT__1, which returns a percentage only.

 

I have tried all the different units in your list without success.

 

Am I missing the obvious again?

 

Regards,

 

David
